tests/std/tests/P0896R4_views_join/test.cpp

```diff
@@ -132,7 +132,12 @@ constexpr bool test_one(Outer&& rng, Expected&& expected) {
 
         // Validate deduction guide
         same_as<R> auto r = join_view{forward<Outer>(rng)};
-        assert(ranges::equal(r, expected));
+#ifdef __cpp_lib_constexpr_dynamic_alloc

```

Comment:
is this ifdef in the right direction?

I'd expect "if we have dynalloc then do the thing, otherwise only do the thing if we're not constexpr" this is "if we have dynalloc then only do the thing if we're not constexpr, otherwise do the thing

Reply:
Good catch: it's wrong, but the test doesn't notice because we don't have coverage from a compiler that doesn't support constexpr dynamic allocation but _does_ support concepts.